cmd/opm/alpha/generate/generate.go

```diff
@@ -0,0 +1,117 @@
+package generate
+
+import (
+	"context"
+	"fmt"
+	"github.com/containerd/containerd/platforms"
+	"github.com/operator-framework/operator-registry/pkg/image"
+	"github.com/operator-framework/operator-registry/pkg/image/containerdregistry"
+	"github.com/operator-framework/operator-registry/pkg/image/containerdregistry/builder"
+	"github.com/spf13/cobra"
+	"strings"
+)
+
+func NewCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "generate <path to declarative config directory>",
+		Short: "Generate an oci-archive index image for a declarative config directory",
+		Args:  cobra.MinimumNArgs(1),
+		RunE: generate,
+	}
+	cmd.Flags().StringP("binary-image", "i", "quay.io/joelanford/opm:a8c80ac9", "container image for on-image `opm` command")
+	cmd.Flags().StringP("tag", "t", "", "custom tag for container image being built")
+	cmd.Flags().Bool("generate", false, "generate a dockerfile")
+	cmd.Flags().StringP("out-dockerfile", "d", "", "if generating the dockerfile, this flag is used to (optionally) specify a dockerfile name")
+	cmd.Flags().StringP("output", "o", "", "name of the output archive")
+	cmd.Flags().StringSlice("labels", nil, "list of label key=value pairs to apply on the index image")
```

Comment:
What do you think about making the CLI a light wrapper around another action API, similar to  the [render command](https://github.com/operator-framework/operator-registry/blob/9a4d16f18f26bf81d3bdbffdba2773481a5a95d1/cmd/opm/render/cmd.go#L18) and [`action.Render`](https://github.com/operator-framework/operator-registry/blob/9a4d16f18f26bf81d3bdbffdba2773481a5a95d1/internal/action/render.go#L52-L56)?

This way we can make a nice Go API for building OCI archives for declarative config indexes.